### ISSUE TITLE
Horizontal scrolling not convenient

### ISSUE THREAD

**Issue by ebuildy** [mozilla-frontend-infra/react-lazylog#21]
When log have some long lines, the horizontal scroll bar is showing at the bottom, so we must scroll down to scroll hozirontaly.

Not really handy with ton of rows.

**Comment by camd** [comment]
I'm hitting this, too.  On a mac with trackpad, you can two-finger drag to scroll right.  I think on windows you can click and drag, perhaps?  But having the horizontal scrollbar visible would help this a lot.

**Comment by helfi92** [comment]
Give the following a try:

```jsx
const containerStyle = {
  width: 'auto',
  maxWidth: 'initial',
  overflow: 'initial',
};

// ...

<LazyLog containerStyle={containerStyle} ... />
```

**Comment by helfi92** [comment]
Let me know if it works for you. I can create a PR to bring this fix upstream.

**Comment by camd** [comment]
@helfi92 : When I made these settings, I wasn't able to scroll horizontally at all anymore, even with the trackpad 2-finger scroll.  My component is defined with the ``containerStyle`` above and:

              <LazyLog
                url={rawLogUrl}
                stream
                scrollToLine={highlight ? highlight[0] : 0}
                highlight={highlight}
                selectableLines
                onHighlight={this.onHighlight}
                onLoad={() => this.scrollHighlightToTop(highlight)}
                highlightLineClassName="yellow-highlight"
                rowHeight={13}
                extraLines={3}
                containerStyle={containerStyle}
              />

**Comment by camd** [comment]
FWIW: If I set ``overflow: 'scroll'`` instead, then I can scroll both vertically and horizontally.  But the horizontal scrollbar scrolls off-screen when you scroll vertically.  So it would seem that horizontal scrolling is handled by one container, which is contained by the one that handles vertical scrolling?

**Comment by helfi92** [comment]
@camd, could you point me to the code in treeherder? I'll give it a try on my machine.

**Comment by camd** [comment]
Sure, thanks!  Here you go:  https://github.com/mozilla/treeherder/blob/3c02823ea886a55e06e1e25aff42173f71a7a273/ui/logviewer/App.jsx#L232

With the repo downloaded, just a ``yarn && yarn start`` will get it going for you.  Thanks again.  :)

**Comment by helfi92** [comment]
@camd 

With

```diff
diff --git a/ui/logviewer/App.jsx b/ui/logviewer/App.jsx
index b1be00496..8ae8a0f9f 100644
--- a/ui/logviewer/App.jsx
+++ b/ui/logviewer/App.jsx
@@ -203,6 +203,12 @@ class App extends React.PureComponent {
       },
     ];
 
+    const containerStyle = {
+      width: 'auto',
+      maxWidth: 'initial',
+      overflow: 'initial',
+    };
+
     return (
       <div className="d-flex flex-column body-logviewer h-100">
         <Navigation
@@ -239,6 +245,7 @@ class App extends React.PureComponent {
                 highlightLineClassName="yellow-highlight"
                 rowHeight={13}
                 extraLines={3}
+                containerStyle={containerStyle}
               />
             </div>
           </div>
```

I was able to scroll horizontally. For example, clicking on the mouse wheel shows:

![screen shot 2019-02-13 at 9 01 17 am](https://user-images.githubusercontent.com/3766511/52717008-14992680-2f6e-11e9-833d-5b171e4921e6.png)

Note the all direction scrolling will only work when the visible text is long. When the visible text does not overflow, the scroll will only work vertically. That's because the LazyLog component only render one section at a time to maintain performance.

**Comment by camd** [comment]
@helfi92 : Yeah, horizontal scrolling is possible.  The issue is that the scrollbar is not visible, so I think that's confusing for some folks.

**Comment by helfi92** [comment]
> The issue is that the scrollbar is not visible, so I think that's confusing for some folks.

You probably don't want to have a scroller that changes width while scrolling down (since the scroller width will vary with the current visible text and in some cases there won't be any scroller if all visible text fits in the screen). That would affect the UX imo.

**Comment by camd** [comment]
That's a fair point.  OK.  So I suppose this is likely a WONTFIX, by Bugzilla nomenclature.

**Comment by whimboo** [comment]
For me it's a huge draw back, given that the mouse only allows to scroll vertically. By having no scrollbar I have to click and move right, which would select the underlying text.

**Comment by timdudman** [comment]
> Give the following a try:
> 
> ```js-jsx
> const containerStyle = {
>   width: 'auto',
>   maxWidth: 'initial',
>   overflow: 'initial',
> };
> 
> // ...
> 
> <LazyLog containerStyle={containerStyle} ... />
> ```

This works except when the lines haven't yet reached the bottom of the screen. Is there any workaround for that?

**Comment by helfi92** [comment]
> This works except when the lines haven't yet reached the bottom of the screen. Is there any workaround for that?

@timdudman, this has been resolved in react-lazylog@4.4.3.

![Screen Shot 2019-11-22 at 10 38 17 AM](https://user-images.githubusercontent.com/3766511/69439074-381aaa00-0d14-11ea-8fd5-983ad1b66ac9.png)

@camd If you update to the latest version, then you don't need to use the `containerStyle` hack from https://github.com/mozilla-frontend-infra/react-lazylog/issues/21#issuecomment-463211264

**Comment by timdudman** [comment]
Looks good @helfi92.
